Sources/Lifetime.swift

```diff
@@ -39,6 +39,58 @@ public final class Lifetime {
 		self.init(ended: token.ended)
 	}
 
+	/// Create a lifetime that is an OR of `self` and `other`.
+	///
+	/// - parameters:
+	///   - other: The lifetime to be composed with.
+	///
+	/// - returns:
+	///   An OR lifetime.
+	public func or(_ other: Lifetime) -> Lifetime {
+		return Lifetime(ended: ended.zip(with: other.ended).map { _ in })
+	}
+
+	/// Create a lifetime that is an AND of `self` and `other`.
```

Comment:
It wasn't. It interrupts when the object is freed *at the time the producer starts*. But it never stops once it is started. It is quite obvious as there is no reference to `lifetime` in `notifications`.